src/wallet/api/wallet.cpp

```diff
@@ -2331,22 +2331,28 @@ bool WalletImpl::isKeysFileLocked()
     return m_wallet->is_keys_file_locked();
 }
 
-PendingTransaction* WalletImpl::stakePending(const std::string& sn_key_str, const std::string& address_str, const std::string& amount_str)
+PendingTransaction* WalletImpl::stakePending(const std::string& sn_key_str, const std::string& address_str, const std::string& amount_str, std::string& error_msg)
 {
   crypto::public_key sn_key;
-  if (!epee::string_tools::hex_to_pod(sn_key_str, sn_key)) {
+  if (!epee::string_tools::hex_to_pod(sn_key_str, sn_key))
+  {
+    error_msg = "Failed to parse service node pubkey";
     LOG_ERROR("failed to parse service node pubkey");
     return nullptr;
   }
 
   cryptonote::address_parse_info addr_info;
-  if (!cryptonote::get_account_address_from_str_or_url(addr_info, m_wallet->nettype(), address_str)) {
+  if (!cryptonote::get_account_address_from_str_or_url(addr_info, m_wallet->nettype(), address_str))
+  {
+    error_msg = "Failed to parse the contributor's address";
     LOG_ERROR("failed to parse address");
     return nullptr;
   }
 
   uint64_t amount;
-  if (!cryptonote::parse_amount(amount, amount_str)) {
+  if (!cryptonote::parse_amount(amount, amount_str))
+  {
+    error_msg = "Incorrect amount";
```

Comment:
Make error msg the contents of the LOG_ERROR() then log the error_msg?

Reply:
Changed and rebased